src/Umbraco.Core/Persistence/Repositories/Implement/ContentRepositoryBase.cs

```diff
@@ -798,5 +818,65 @@ public virtual IEnumerable<TEntity> GetRecycleBin()
         }
 
         #endregion
+
+        protected void PersistRelations(TEntity entity)
+        {
+            var trackedRelations = new List<UmbracoEntityReference>();
+
+            foreach (var p in entity.Properties)
+            {
+                if (!PropertyEditors.TryGet(p.PropertyType.PropertyEditorAlias, out var editor)) continue;
+                var valueEditor = editor.GetValueEditor();
+                if (!(valueEditor is IDataValueReference reference)) continue;
+
+                //TODO: Support variants/segments! This is not required for this initial prototype which is why there is a check here
+                if (!p.PropertyType.VariesByNothing()) continue;
+
+                var val = p.GetValue(); // get the invariant value
+                var refs = reference.GetReferences(val);
+                trackedRelations.AddRange(refs);
+            }
+
+            //First delete all auto-relations for this entity
+            RelationRepository.DeleteByParent(entity.Id, Constants.Conventions.RelationTypes.AutomaticRelationTypes);
+
+            if (trackedRelations.Count == 0) return;
+
+            var udiToGuids = trackedRelations.Select(x => x.Udi as GuidUdi)
```

Comment:
```suggestion
            var udiToGuids = trackedRelations.Select(x => x.Udi as GuidUdi).Distinct()
```

We need to district here, otherwise ToDictionary throws if I have multiple relations to the same item

I tested this by adding both a link to an image and the image itself into the RTE

Reply:
I see we actually need to distinct on trackedRelations, otherwise we have `A duplicate value cannot be inserted into a unique index` error from the database